File: Collections/IOrderedDictionary.vb
Imports System.Collections.Specialized

Namespace Common

    Public Interface IOrderedDictionary(Of TKey, TValue)
        Inherits IOrderedDictionary,
                 IList,
                 IList(Of KeyValuePair(Of TKey, TValue)),
                 IDictionary(Of TKey, TValue),
                 IDictionary

        Overloads Sub Insert(index As Integer, key As TKey, value As TValue)

    End Interface

End Namespace

File: Collections/IOrderedDictionary.cs
using System.Collections.Generic;
using System.Collections;
using System.Collections.Specialized;

namespace Ks.Common
{
        public interface IOrderedDictionary<TKey, TValue> : IOrderedDictionary, IList, IList<KeyValuePair<TKey, TValue>>, IDictionary<TKey, TValue>, IDictionary
        {
            void Insert(int index, TKey key, TValue value);
        }
    }
